examples/src/bin/clipboard_simple.rs

```diff
@@ -109,12 +109,11 @@ fn build_ui(application: &gtk::Application) {
     paste_button.connect_clicked(|_| {
         let clipboard = gtk::Clipboard::get(&gdk::SELECTION_CLIPBOARD);
         clipboard.request_text(|_, t| {
-            if t.is_some() {
-                let t = t.unwrap();
+            if let Some(t) = t {
                 if t.len() >= 4 {
                     GLOBAL.with(|global| {
                         if let Some(ref ui) = *global.borrow() {
-                            ui.button_a1.set_active(t.chars().nth(0).unwrap() == '1');
+                            ui.button_a1.set_active(t.starts_with('1'));
```

Comment:
This is very strange code :) How about

```rust
let t = t.chars().collect::<Vec<_>>();
if t.len() >= 4 {
    [...]
    ui.button_a1.set_active(t[0] == '1');
    [...]
}
```

Reply:
Indexing on strings is a very bad idea. If you have a multi-bytes character as the first one, you'll panic.

Reply:
Yes, read again what I wrote. That's a `Vec<char>` :P

Reply:
Then nevermind. :3